src/util/isArrayLike.ts

```diff
@@ -0,0 +1 @@
+export const isArrayLike = (<T>(x: any): x is ArrayLike<T> => x && typeof x.length === 'number');
```

Comment:
do we have common usage's of this function over codebase except `from`? if not, what about defer to export it once there's usage?